Marlin/src/core/serial_base.h

```diff
@@ -79,11 +79,11 @@ struct SerialBase {
   FORCE_INLINE void write(const uint8_t* buffer, size_t size) { while (size--) write(*buffer++); }
   FORCE_INLINE void print(const char* str)                    { write(str); }
   NO_INLINE void print(char c, int base = 0)               { print((long)c, base); }
-  NO_INLINE void print(unsigned char c, int base = 0)      { print((unsigned long)c, base); }
+  NO_INLINE void print(unsigned char c, int base = DEC)    { print((unsigned long)c, base); }
```

Comment:
This will break code doing `print(uint8_t)` to print a char on platform where `uint8_t` is a `char` (like ARM). It does not seem correct to me. I need to check this...

Reply:
Well char treated as unsigned is not standard. In all system I faced char was always a signed 8 bit value (ASCII uses only 7 bits and all system are derived from such assumption. In my system I said something wrong, uint8_t is unsigned char as it should be (my fix works correctly with loop variables).
Now I don't know what function was previously used to print unsigned char but Marlin was working correctly before your PR then checking original framework call may be the solution

Reply:
You are right. I'm talking about the signedness of `char`. On X86 and probably AVR (not 100% sure), `char == signed char` but on ARM, `char == unsigned char`. Since `uint8_t == unsigned char` on all platform, when you have `print((uint8_t)3)`, your intention is to print the decimal number 3. But on ARM, it'll call `print(char)` to print the character with ASCII value 3 (since uint8_t is a char for this platform).

Reply:
uh that's a problem but I'm wondering how it worked before (G29 T), maybe on platform with char=unsigned char uint8_t is define as something else. If this is the case, replace che unsigned char version getting uint8_t and the game is done

Reply:
Yes, that's the idea.